lib/syskit/network_generation/engine.rb

```diff
@@ -681,52 +689,57 @@ def adapt_existing_deployment(deployment_task, existing_deployment_task)
             # Computes the set of requirement tasks that should be used for
             # deployment within the given plan
             def self.discover_requirement_tasks_from_plan(plan)
-                req_tasks = plan.find_local_tasks(InstanceRequirementsTask)
-                req_tasks = req_tasks.find_all do |req_task|
-                    if req_task.failed? || req_task.pending?
-                        false
-                    elsif (planned_task = req_task.planned_task)
-                        !planned_task.finished? || planned_task.being_repaired?
-                    else
-                        false
-                    end
-                end
-                needed = plan.useful_tasks(with_transactions: false)
-                req_tasks.delete_if do |t|
-                    !needed.include?(t)
-                end
-                req_tasks
+                req_tasks =
+                    plan.find_local_tasks(InstanceRequirementsTask).running
+                req_tasks = req_tasks.find_all do |t|
+                    planned_task = t.planned_task
+                    !planned_task&.finished? || planned_task&.being_repaired?
```

Comment:
`!planned_task&.finished?` == true if planned_task.nil?

Not sure that's really what we want...